client/app/bundles/course/assessment/submission/actions/submissions.js

```diff
@@ -116,23 +116,35 @@ export function sendAssessmentReminderEmail(assessmentId, type) {
   };
 }
 
-export function downloadSubmissions(type) {
+export function downloadSubmissions(type, downloadFormat) {
+  const actions =
+    downloadFormat === 'zip'
+      ? [
+          actionTypes.DOWNLOAD_SUBMISSIONS_FILES_REQUEST,
+          actionTypes.DOWNLOAD_SUBMISSIONS_FILES_SUCCESS,
+          actionTypes.DOWNLOAD_SUBMISSIONS_FILES_FAILURE,
+        ]
+      : [
+          actionTypes.DOWNLOAD_SUBMISSIONS_CSV_REQUEST,
+          actionTypes.DOWNLOAD_SUBMISSIONS_CSV_SUCCESS,
+          actionTypes.DOWNLOAD_SUBMISSIONS_CSV_FAILURE,
+        ];
```

Comment:
Nice! I think this works, but if we want to make it more semantics, we can perhaps do:

```js
{
  request: actionTypes.DOWNLOAD_SUBMISSIONS_FILES_REQUEST,
  success: actionTypes.DOWNLOAD_SUBMISSIONS_FILES_SUCCESS,
  failure: actionTypes.DOWNLOAD_SUBMISSIONS_FILES_FAILURE,
}
```

instead, then `action.request` and so on would be easier to read.